core/extensions/AssemblyLoading/AssemblyLoading/Program.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace AssemblyLoading
+{
+    static class Usage
+    {
+        public const string DefaultContext = "default";
+        public const string CustomContext = "custom";
+
+        public const string AssemblyLoadContextLoad  = "alc-load";
+        public const string AssemblyLoadContextResolvingDefault = "alc-resolving-default";
+        public const string AssemblyLoadContextResolving = "alc-resolving";
+        public const string AppDomainAssemblyResolve= "appdomain-assemblyresolve";
+
+        public static void Print()
+        {
+            const int alignment = -28;
+            Console.WriteLine($@"
+Usage:
+  AssemblyLoading <context> [<extension-point>]
+
+context:
+  {DefaultContext}  Use the default AssemblyLoadContext
+  {CustomContext}   Use a new custom AssemblyLoadContext
+
+extension-point:
+  {AssemblyLoadContextLoad,alignment} AssemblyLoadContext.Load loads the assembly - if <context> is 'default', this has no effect
+  {AssemblyLoadContextResolvingDefault,alignment} AssemblyLoadContext.Resolving event handler for the default AssemblyLoadContext loads the assembly
+  {AssemblyLoadContextResolving,alignment} AssemblyLoadContext.Resolving event handler loads the assembly - if <context> is 'default', this is equivalent to {AssemblyLoadContextResolvingDefault}
+  {AppDomainAssemblyResolve,alignment} AppDomain.AssemblyResolve event handler loads the assembly
+
+The <extension-point> specifies which extension point will successfully load the library.
+If not specified, the library will fail to load.");
+        }
+
+        public static bool IsValidExtensionPoint(string value) => value switch
+        {
+            AssemblyLoadContextLoad
+                or AssemblyLoadContextResolving
+                or AssemblyLoadContextResolvingDefault
+                or AppDomainAssemblyResolve => true,
+            _ => false
+        };
+    }
+
+    class Program
+    {
+        private const string libraryName = "MyLibrary";
+        private static string libraryPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), $"{libraryName}.dll");
```

Comment:
```suggestion
        private static string libraryPath = Path.Combine(AppContext.BaseDirectory, $"{libraryName}.dll");
```

Since we're now building single-file - this should work with it :wink: